app/main.py

```diff
@@ -1,8 +1,22 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    list_with_person_inst = [
+        Person(name=person["name"], age=person["age"])
+        for person in people
+    ]
+    for human in people:
+        person = Person.people[human["name"]]
+        if "wife" in human and human["wife"] is not None:
+            person.wife = Person.people[human["wife"]]
```

Comment:
You are dynamically adding the `wife` attribute to the `Person` instance. This can lead to potential attribute errors if you try to access `wife` on a `Person` instance that doesn't have it. Consider initializing `wife` and `husband` attributes in the `Person` class constructor with default values (e.g., `None`) to avoid such issues.